pac4j-jwt/src/main/java/org/pac4j/jwt/credentials/authenticator/JwtAuthenticator.java

```diff
@@ -114,8 +138,10 @@ public void validate(final TokenCredentials credentials) {
             } else {
                 throw new TechnicalException("unsupported unsecured jwt");
             }
+            
+    		JWSVerifier verifier = factory.createJWSVerifier(signedJWT.getHeader(), key);
```

Comment:
It's a good idea indeed to create the verifier from the signedJWT header, the problem is that the key must be of the right format.